src/overlays/actors/ovl_Bg_F40_Swlift/z_bg_f40_swlift.c

```diff
@@ -28,24 +31,83 @@ ActorInit Bg_F40_Swlift_InitVars = {
     (ActorFunc)BgF40Swlift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8096F540[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 550, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 5000, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgF40Swlift_Init(Actor* thisx, PlayState* play) {
+    BgF40Swlift* this = THIS;
+    s32 index;
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
+    index = BG_F40_SWLIFT_GET_INDEX(thisx);
+    if ((index < 0) || (index >= 5)) { //! @bug An index greater than 3 will cause an out of bounds array access.
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        sHeights[index] = this->dyna.actor.world.pos.y;
+        sSwitchFlags[index] = BG_F40_SWLIFT_GET_SWITCHFLAG(thisx);
+        if (index != 0) {
+            Actor_Kill(&this->dyna.actor);
+        } else {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerVerticallyOscillatingPlatformCol);
+            this->dyna.actor.params = 0;
+        }
```

Comment:
```suggestion
        if (index != 0) {
            Actor_Kill(&this->dyna.actor);
            return;
        }

        DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerVerticallyOscillatingPlatformCol);
        this->dyna.actor.params = 0;
```
We try to early-return for `Actor_Kill`